fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -79,10 +77,106 @@ static_assert((ROCKSDB_MAJOR == 6 && ROCKSDB_MINOR == 27) ? ROCKSDB_PATCH >= 3 :
 namespace {
 using rocksdb::BackgroundErrorReason;
 
-struct SharedRocksDBState {
-	bool closing = false;
+class SharedRocksDBState {
+public:
+	SharedRocksDBState();
+
+	void setClosing() { this->closing = true; }
```

Comment:
Is this just used for rocksDBMetricLogger? The metrics future is reset anyways and that will break the metricsLogger loop right?

Reply:
In some rare case, there is a race between delete db; and the cancellation of rocksDBMetricLogger ACTOR, `closing` is used to synchronize them.